scripts/main.py

```diff
@@ -46,15 +49,74 @@ def run(config: dict, cache: str):
     incident_model = LinearIncidentUptakeModel().fit(
         incident_train_data, grouping_factors
     )
-    cumulative_projections = incident_model.predict(
-        config["timeframe"]["start"],
-        config["timeframe"]["end"],
-        config["timeframe"]["interval"],
-        grouping_factors,
-    )
-    print(cumulative_projections)
-    incident_projections = cumulative_projections.to_incident(grouping_factors)
-    print(incident_projections)
+
+    option = config["option"]
+
+    if option == "projection":
+        cumulative_projections = incident_model.predict(
+            config["timeframe"]["start"],
+            config["timeframe"]["end"],
+            config["timeframe"]["interval"],
+            grouping_factors,
+        )
+        print(cumulative_projections)
+        incident_projections = cumulative_projections.to_incident(grouping_factors)
+        print(incident_projections)
+
+    elif option == "evaluation":
+        # Make projections sequentially
+        interval_str = config["timeframe"]["interval"]
+        interval_number = int(interval_str[0])
+
+        dates = pl.date_range(
+            config["timeframe"]["start"],
+            config["timeframe"]["end"] - dt.timedelta(interval_number),
+            config["timeframe"]["interval"],
+            eager=True,
+        )
+
+        scores = pl.DataFrame(
+            schema={
+                "forecast_start": pl.Date,
+                "forecast_end": pl.Date,
+                "score": pl.Float64,
+                "type": pl.String,
+            }
+        )
+
+        for date in dates:
+            # Generate cumulative predictions
+            cumulative_projections = incident_model.predict(
+                date,
+                config["timeframe"]["end"],
+                config["timeframe"]["interval"],
+                grouping_factors,
+            )
+
+            pred = iup.PointForecast(
+                cumulative_projections.to_incident(grouping_factors)
+            )
+            test_data = iup.IncidentUptakeData(
+                iup.IncidentUptakeData.split_train_test(
+                    incident_data, date, "test"
+                ).filter(pl.col("date") <= config["timeframe"]["end"])
+            )
+
+            if config["metric"] == "mspe":
```

Comment:
Rather than these if-else's, you could do something like:

```python
score_fun = getattr(eval, config['metric'])
score = eval.score(test_data, pred, score_fun)
```

Essentially you grab the function based on its string name

Reply:
See [b127d43](https://github.com/CDCgov/cfa-immunization-uptake-projection/pull/82/commits/b127d43fe50e80b1f31a684d7a3c6e0596265c17)